### ISSUE TITLE
Cannot use workflow config parser without pegasus installed

### ISSUE THREAD

**Issue by cdcapano** [ligo-cbc/pycbc#883]
If I try to import pycbc.workflow.WorkflowConfigParser on my laptop I get an import error because I do not have pegasus installed. This is a problem if I want to run pycbc_mcmc on my laptop, which uses WorkflowConfigParser. I suppose we could just have pycbc_mcmc use ConfigParser, however, this seems like a bug in the workflow module. There is nothing in WorkflowConfigParser that requires pegasus; it just offers fancier config parsing than the basic ConfigParser. Pegasus should not be required in order to use it.

**Comment by spxiwh** [comment]
Try:

```
from pycbc.workflow.configuration import WorkflowConfigParser
```

?

**Comment by cdcapano** [comment]
That doesn't work. The problem is in workflow's __init__.py, a bunch of modules are imported that try to import pegasus. Even if you try to import the class directly, workflow's __init__.py gets called. Also, WorkflowConfigParser uses resolve_url which lives in workflow.core; workflow.core imports pegasus.

I think __init__.py needs to adjusted so that modules that require Pegasus are stuck in a try/except block, with everything else (like configuration.py) outside of that block. Also, resolve_url needs to be moved to another module that does not require pegasus (configuration.py?).

**Comment by duncan-brown** [comment]
You missed this step from the install instructions:
```
pip install http://download.pegasus.isi.edu/pegasus/4.6.1/pegasus-python-source-4.6.1.tar.gz
```
(Note the instructions are out of date as it gives an old version, but the part that gets imported hasn't changed).

**Comment by ahnitz** [comment]
@duncan-brown This is a legitimate issue. Pegasus *should not* be a requirement for executable use.

**Comment by ahnitz** [comment]
@cdcapano What we should do is hide some of the imports better in this case, so some of the configparser utilities can be used.

**Comment by duncan-brown** [comment]
@ahnitz ok, didn't read the thread properly. Sounds like this is in the config parsing code, so I'll assign to you.

**Comment by cdcapano** [comment]
@duncan-brown BTW, I tried installing Pegasus as you suggested, but it fails on my system. The problem is Apple no longer includes OpenSSL headers in El Capitan. Apparently the only work around is to use the homebrew installation of OpenSSL; see:
https://jira.isi.edu/browse/PM-997
However, I tried installing OpenSSL and it still didn't work. Not sure what I did wrong. Either way, installing Pegasus on a Mac is currently a pain.
